pkg/cmd/roachtest/spec/machine_type.go

```diff
@@ -61,27 +100,45 @@ func AWSMachineType(cpus int, mem MemPerCPU, arch vm.CPUArch) (string, vm.CPUArc
 		panic(fmt.Sprintf("no aws machine type with %d cpus", cpus))
 	}
 
-	// There is no m7g.24xlarge, fall back to m6i.24xlarge.
-	if family == "m7g" && size == "24xlarge" {
-		family = "m6i"
+	// There is no m7g.24xlarge, fall back to (c|m|r)6i.24xlarge.
+	if selectedArch == vm.ArchAMD64 && size == "24xlarge" {
+		switch mem {
+		case Auto:
+			family = "c6i"
+		case Standard:
+			family = "m6i"
+		case High:
+			family = "r6i"
+
+		}
+		if shouldSupportLocalSSD {
+			family = family + "d"
```

Comment:
Can we simplify this function by only having one check at the end for `if shouldSupportLocalSSH { family += "d"  }`?